app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,34 @@
+class CommentsController < ApplicationController
+  before_action :set_user_and_post, only: %i[new create]
+
+  def new
+    @comment = @post.comments.new
+  end
+
+  def create
+    @comment = @post.comments.new(comment_params)
+    @comment.user = @user
+
+    puts "User: #{@user.inspect}"
+    puts "Post: #{@post.inspect}"
+    puts "Comment: #{@comment.inspect}"
+
+    if @comment.save
+      flash[:success] = 'Comment was successfully created.'
+      redirect_to user_post_path(@user, @post)
+    else
+      render :new
+    end
+  end
```

Comment:
- [ ] @ometman Great work adding a method to add new comments, however, we are supposed to use the current_user method created on the application controller to create a new comment on behalf of the current user, kindly consider using the method when creating the comment.
![image](https://github.com/ometman/blogtalk/assets/61408860/8049b880-2e3a-48ca-b275-356831d62665)

Reply:
Fixed. Thanks